### PR TITLE
Add passport local and facebook strategies authorization

### PR DESCRIPTION
**Pull request by MaratOM**

### CODE REVIEW

File: lesson5/models/user.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  password: {
+    type: String,
+  },
+  facebookId: {
+    type: String,
+  },
+})
+
+userSchema.statics.findOrCreate = async function(service, userData) {
```
Comment (ubcent) [inline lesson5/models/user.js:18]: It seems like we're having already this method

**Review by ubcent — APPROVED** [review]

File: lesson5/models/user.js (hunk 1 above)
Comment (MaratOM) [reply lesson5/models/user.js:18]: It's seems than I have older version of mongo, that doesn't have new methods. Should update it>

Comment (ubcent) [reply lesson5/models/user.js:18]: Do you mean mongoose?

Comment (MaratOM) [reply lesson5/models/user.js:18]: You're right, it's mongoose, but needs additional plugin as i found out https://github.com/drudge/mongoose-findorcreate
Without it node doesn't see the function, says function not found.

Comment (ubcent) [reply lesson5/models/user.js:18]: Actually, I mean [this](http://mongoosejs.com/docs/api.html#findoneandupdate_findOneAndUpdate). You need to specify the "upsert" parameter as well

Comment (MaratOM) [reply lesson5/models/user.js:18]: Thanks, I will discover it.

**Pull request merged**